### DISCUSSION TITLE
Fais-cpu fails to build with swig

### DISCUSSION THREAD

**Discussion by sheneeb** [facebookresearch/faiss#3599] [General]
pip install faiss-cpu
                                                  Collecting faiss-cpu
  Using cached faiss-cpu-1.8.0.tar.gz (63 kB)                                   Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done                                Requirement already satisfied: numpy in /data/data/com.termux/files/usr/lib/python3.11/site-packages (from faiss-cpu) (1.23.2)
Building wheels for collected packages: faiss-cpu                               Building wheel for faiss-cpu (pyproject.toml) ... error
  error: subprocess-exited-with-error
                                                                                × Building wheel for faiss-cpu (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [12 lines of output]                                                          running bdist_wheel
      running build
      running build_py                                                              running build_ext
      building 'faiss._swigfaiss' extension
      swigging faiss/faiss/python/swigfaiss.i to faiss/faiss/python/swigfaiss_wrap.cpp                                                                            swig -python -c++ -Doverride= -doxygen -Ifaiss -I/data/data/com.termux/files/usr/tmp/pip-build-env-8fqadydt/overlay/lib/python3.11/site-packages/numpy/core/include -Ifaiss -I/usr/local/include -DSWIGWORDSIZE64 -o faiss/faiss/python/swigfaiss_wrap.cpp faiss/faiss/python/swigfaiss.i
      Traceback (most recent call last):                                              File "/data/data/com.termux/files/usr/bin/swig", line 5, in <module>
          from swig import swig
      ModuleNotFoundError: No module named 'swig'                                   error: command '/data/data/com.termux/files/usr/bin/swig' failed with exit code 1
      [end of output]                                                         
  note: This error originates from a subprocess, and is likely not a problem with pip.                                                                        ERROR: Failed building wheel for faiss-cpu                                  Failed to build faiss-cpu
ERROR: Could not build wheels for faiss-cpu, which is required to install pyproject.toml-based projects

**Comment by mdouze** [comment]
please follow the install instructions

**Comment by sheneeb** [comment]
The installation is not specific to termux on android which only supports pip, any help to get out installed through  pip on android

**Comment by vorj** [comment]
@sheneeb faiss on pip (`faiss-cpu` and `faiss-gpu`) is hosted unofficially by [kyamagu/faiss-wheels](https://github.com/kyamagu/faiss-wheels), so it is more appropriate to report the problem there rather than here.

**Reply by sheneeb** [reply]
There is a mention of installation on raspberry pi any way to get hold of that
